src/libutil/util.cc

```diff
@@ -54,6 +54,13 @@ std::optional<std::string> getEnv(const std::string & key)
     return std::string(value);
 }
 
+std::optional<std::string> getEnvNonEmpty(const std::string & key)
+{
+    char * value = getenv(key.c_str());
+    if (!value) return {};
+    auto str = std::string(value);
+    return !str.empty() ? std::optional<std::string>{str} : std::nullopt;
+}
```

Comment:
I guess this could be expressed in terms of `getEnv()`? I.e. something like
```
    auto value = getEnv(key);
    return value == "" ? std::nullopt : value;
```